### PR TITLE
MNT-22696 Replace renditions via REST API

### PR DESCRIPTION
**Pull request by SaraAspery**
Added new endpoints:
```
DELETE /nodes/{nodeId}/renditions/{renditionId}
DELETE /nodes/{nodeId}/versions/{versionId}/renditions/{renditionId}
```

### CODE REVIEW

File: src/main/webapp/definitions/alfresco-core.yaml [hunk 1]
```diff
@@ -2123,6 +2123,39 @@ paths:
...
             $ref: '#/definitions/Error'
+    delete:
+      x-alfresco-since: "7.1.1"
+      tags:
+        - renditions
+      summary: Delete rendition
+      description: |
+        **Note:** this endpoint is available in Alfresco 7.1.1 and newer versions.
+
+        Delete the rendition identified by **renditionId** of file **nodeId**.
```
Comment (alandavis) [inline src/main/webapp/definitions/alfresco-core.yaml:2134]: "file" does not sound right. Use what is used more generally to refer to nodeIds. I think you may have copied it from other 'renditions' descriptions. They look wrong too, but not suggesting fixing them at this time. I would suggest looking a the descriptions under the 'nodes' section.

File: src/main/webapp/definitions/alfresco-core.yaml [hunk 2]
```diff
@@ -3106,6 +3139,40 @@ paths:
...
+    delete:
+      x-alfresco-since: "7.1.1"
+      tags:
+        - versions
+      summary: Delete rendition for a file version
+      description: |
+        **Note:** this endpoint is available in Alfresco 7.1.1 and newer versions.
+
+        Delete the rendition for **renditionId** of version of file **nodeId** and **versionId**.
+
```
Comment (alandavis) [inline src/main/webapp/definitions/alfresco-core.yaml:3151]: same here

File: src/main/webapp/definitions/alfresco-core.yaml [hunk 3]
```diff
@@ -2123,6 +2123,39 @@ paths:
...
+      operationId: deleteRendition
+      parameters:
+        - $ref: '#/parameters/nodeIdParam'
+        - $ref: '#/parameters/renditionIdParam'
+      responses:
+        '204':
+          description: Successful response
+        '400':
+          description: |
+            Invalid parameter: **nodeId** is not a valid format, or is not a file, or **renditionId** is invalid
```
Comment (alandavis) [inline src/main/webapp/definitions/alfresco-core.yaml:2147]: Same here - is not a file sounds like an upload failure.

File: src/main/webapp/definitions/alfresco-core.yaml [hunk 4]
```diff
@@ -3106,6 +3139,40 @@ paths:
...
+      parameters:
+        - $ref: '#/parameters/nodeIdParam'
+        - $ref: '#/parameters/versionIdParam'
+        - $ref: '#/parameters/renditionIdParam'
+      responses:
+        '204':
+          description: Successful response
+        '400':
+          description: |
+            Invalid parameter: **nodeId** is not a valid format, or is not a file, or **renditionId** is invalid
```
Comment (alandavis) [inline src/main/webapp/definitions/alfresco-core.yaml:3164]: same here

**Review by alandavis — CHANGES_REQUESTED** [review]

File: src/main/webapp/definitions/alfresco-core.yaml (hunk 1 above)
Comment (SaraAspery) [reply src/main/webapp/definitions/alfresco-core.yaml:2134]: How about "Delete the rendition identified by **renditionId** of **nodeId**."?

File: src/main/webapp/definitions/alfresco-core.yaml (hunk 3 above)
Comment (SaraAspery) [reply src/main/webapp/definitions/alfresco-core.yaml:2147]: done

File: src/main/webapp/definitions/alfresco-core.yaml (hunk 4 above)
Comment (SaraAspery) [reply src/main/webapp/definitions/alfresco-core.yaml]: done

**Review by alandavis — APPROVED** [review]

**Pull request merged**